iwyu_ast_util.cc

```diff
@@ -951,13 +953,57 @@ bool IsDefaultNewOrDelete(const FunctionDecl* decl,
       !StartsWith(decl_name, "operator delete"))
     return false;
 
-  // Placement-new/delete has 2 args, second is void*.  The only other
-  // 2-arg overloads of new/delete in <new> take a const nothrow_t&.
-  if (decl->getNumParams() == 2 &&
-      !decl->getParamDecl(1)->getType().isConstQualified())
+  // The following variants of operator new[1] are implicitly defined in every
```

Comment:
I wonder if the initial string-checking at the beginning of this function is even necessary any more?

Reply:
Probably not, no. I've removed it.